lib/chef/tidy_substitutions.rb

```diff
@@ -68,7 +62,7 @@ def run_substitutions
             @data[entry][glob].each do |substitution|
               search = Regexp.new(substitution["pattern"])
               replace = substitution["replace"].dup
-              replace.gsub!(/\!COOKBOOK_VERSION\!/) { |_m| "'" + cookbook_version_from_path(file) + "'" }
+              replace.gsub!(/\!COOKBOOK_VERSION\!/) { |_m| "'" + @tidy.cookbook_version_from_path(file) + "'" }
```

Comment:
This has the same issue where `cookbook_version_from_path` expects the parameter to be a file, although in this case, it's possible for the `file` in question to be deeply nested inside the cookbook and `File.dirname(file)` won't necessarily work.

Reply:
I updated cookbook_version_from_path to be able to handle files nested within a cookbook. It's working for me locally in testing